components/scripts/lib/build-scan-parse.sh

```diff
@@ -25,38 +23,46 @@ executed_cacheable_duration=()
 executed_not_cacheable_num_tasks=()
 executed_not_cacheable_duration=()
 
-# Build duration metrics
-build_time=()
-serialization_factors=()
-
+# Build time metrics
 initial_build_time=""
 instant_savings=""
 instant_savings_build_time=""
 pending_savings=""
 pending_savings_build_time=""
 
-# shellcheck disable=SC2034 # not all scripts use all of the fetched data
-parse_build_scan_csv() {
-  # This isn't the most robust way to read a CSV,
-  # but we control the CSV so we don't have to worry about various CSV edge cases
+# Other build metrics
+serialization_factors=()
 
-  local header_row_read run_num
-  local build_scan_csv="$1"
-  local build_cache_metrics_only="$2"
+parse_single_build_scan() {
+  local build_scan_data="$1"
 
-  debug "Raw build scan data"
-  debug "---------------------------"
-  debug "${build_scan_csv}"
-  debug ""
+  local build_scan_rows
+  IFS=$'\n' read -rd '' -a build_scan_rows <<< "$build_scan_data"
 
-  header_row_read=false
+  parse_build_scan_row all_data "${build_scan_rows[1]}"
+}
 
-  while IFS=, read -r run_num field_1 field_2 field_3 field_4 field_5 field_6 field_7 field_8 field_9 field_10 field_11 field_12 field_13 field_14 field_15 field_16 field_17 field_18 field_19 field_20 field_21; do
-    if [[ "$header_row_read" == "false" ]]; then
-      header_row_read=true
-      continue;
-    fi
+parse_build_scans_and_build_time_metrics() {
+  local build_cache_metrics_only="$1"
+  local build_scan_data="$2"
+
+  local build_scan_rows
+  IFS=$'\n' read -rd '' -a build_scan_rows <<< "$build_scan_data"
```

Comment:
This turned out rather nice.  I wonder if it makes sense to add a comment describing what this does (for those who are less familiar with bash).

Reply:
That's a good idea. I've already forgotten what it does. 🙃

Reply:
I have addressed this in #388